### PR TITLE
APP-63 Fix datacollection demo build

### PR DESCRIPTION
**Pull request by ohamov**

### CODE REVIEW

File: datacollectiondemo/source/c/build.sh [hunk 1]
```diff
@@ -54,54 +51,23 @@ function select_arch {
...
-    cd $PROJECT_HOME &&
-    mkdir -p "$PROJECT_HOME/$BUILD_DIR" &&
-    cp "$KAA_LIB_PATH/$BUILD_DIR/"libkaa* "$PROJECT_HOME/$BUILD_DIR/" &&
-    cd $BUILD_DIR &&
-    cmake -DAPP_NAME=$APP_NAME -DKAA_PLATFORM=$KAA_ARCH $KAA_TOOLCHAIN_PATH_SDK ..
+    mkdir -p "$PROJECT_HOME/$BUILD_DIR"
+    cd $BUILD_DIR
+    cmake -DKAA_PLATFORM=$KAA_ARCH \
+          -DWITH_EXTENSION_EVENT=0 \
+          -DWITH_EXTENSION_CONFIGURATION=0 \
```
Comment (RostakaGmfun) [inline datacollectiondemo/source/c/build.sh]: How about disabling notification extension as well?

File: datacollectiondemo/source/c/build.sh [hunk 2]
```diff
@@ -54,54 +51,23 @@ function select_arch {
...
+          -DKAA_MAX_LOG_LEVEL=3 \
+          -DKAA_PLATFORM=$KAA_ARCH $KAA_TOOLCHAIN_PATH_SDK ..
```
Comment (RostakaGmfun) [inline datacollectiondemo/source/c/build.sh]: KAA_PLATFORM is already specified a few lines above

**Comment by forGGe** [comment]
+

File: datacollectiondemo/source/c/build.sh [hunk 3]
```diff
@@ -54,54 +51,24 @@ function select_arch {
...
-function build_app {
-    cd $PROJECT_HOME &&
-    mkdir -p "$PROJECT_HOME/$BUILD_DIR" &&
-    cp "$KAA_LIB_PATH/$BUILD_DIR/"libkaa* "$PROJECT_HOME/$BUILD_DIR/" &&
-    cd $BUILD_DIR &&
-    cmake -DAPP_NAME=$APP_NAME -DKAA_PLATFORM=$KAA_ARCH $KAA_TOOLCHAIN_PATH_SDK ..
+    mkdir -p "$PROJECT_HOME/$BUILD_DIR"
+    cd $BUILD_DIR
+    cmake -DKAA_PLATFORM=$KAA_ARCH \
+          -DWITH_EXTENSION_EVENT=0 \
```
Comment (rasendubi) [inline datacollectiondemo/source/c/build.sh]: Use `OFF` here.

File: datacollectiondemo/source/c/CMakeLists.txt [hunk 4]
```diff
@@ -15,23 +15,31 @@
...
+               )
+execute_process(COMMAND sh -c "tar -zxf kaa-c*.tar.gz -C ${KAA_SDK_DIR}"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa"
+               )
+
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g -Wall -Wextra")
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src
-                    ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/src
+                    ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/${KAA_SDK_DIR}/src
```
Comment (Dubland) [inline datacollectiondemo/source/c/CMakeLists.txt]: You shouldn't include SDK directory directly.

File: datacollectiondemo/source/c/CMakeLists.txt [hunk 5]
```diff
@@ -15,23 +15,31 @@
...
-enable_language(C)
+project (C-SDK-sample LANGUAGES C)
 
 find_package (OpenSSL REQUIRED)
 
+set(APP_NAME "demo_client")
+set(KAA_SDK_DIR "client-c")
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${KAA_SDK_DIR}
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa"
```
Comment (rasendubi) [inline datacollectiondemo/source/c/CMakeLists.txt]: Why creating a separate directory?

File: datacollectiondemo/source/c/CMakeLists.txt [hunk 6]
```diff
@@ -15,23 +15,31 @@
...
+               )
+execute_process(COMMAND sh -c "tar -zxf kaa-c*.tar.gz -C ${KAA_SDK_DIR}"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa"
+               )
```
Comment (rasendubi) [inline datacollectiondemo/source/c/CMakeLists.txt]: Ideally, you should unpack SDK to the build directory, so yo don't modify source tree.

File: datacollectiondemo/source/c/src/kaa_demo.c [hunk 7]
```diff
@@ -23,7 +23,7 @@
 #include <kaa/kaa_context.h>
 #include <kaa/platform/kaa_client.h>
 #include <kaa/utilities/kaa_log.h>
-#include <kaa/kaa_logging.h>
+#include <extensions/logging/kaa_logging.h>
```
Comment (rasendubi) [inline datacollectiondemo/source/c/src/kaa_demo.c]: Wrong. Should be `<kaa_logging.h>`

File: notificationdemo/source/c/CMakeLists.txt [hunk 8]
```diff
@@ -15,23 +15,31 @@
 #
 
 cmake_minimum_required(VERSION 2.8.8)
-
-project (C-SDK-sample)
-enable_language(C)
+project (C-SDK-sample LANGUAGES C)
```
Comment (rasendubi) [inline notificationdemo/source/c/CMakeLists.txt]: `LANGUAGES` is not supported in cmake 2.8.8. Use `project(C-SDK-sample C)` instead.

File: notificationdemo/source/c/src/kaa_demo.c [hunk 9]
```diff
@@ -23,7 +23,7 @@
 #include <kaa/kaa_error.h>
 #include <kaa/platform/kaa_client.h>
 #include <kaa/utilities/kaa_log.h>
-#include <kaa/kaa_notification_manager.h>
+#include <extensions/notification/kaa_notification_manager.h>
```
Comment (rasendubi) [inline notificationdemo/source/c/src/kaa_demo.c]: Should be `<kaa_notification_manager.h>`.

File: datacollectiondemo/source/c/CMakeLists.txt (hunk 5 above)
Comment (ohamov) [inline datacollectiondemo/source/c/CMakeLists.txt]: Because It's convenient having tar.gz separately from unpacked sdk. It could be easily removed

Comment (rasendubi) [inline datacollectiondemo/source/c/CMakeLists.txt]: See comment below. It's better unpack to the build directory.

File: notificationdemo/source/c/src/kaa_demo.c (hunk 9 above)
Comment (ohamov) [inline notificationdemo/source/c/src/kaa_demo.c]: Done

File: notificationdemo/source/c/CMakeLists.txt (hunk 8 above)
Comment (ohamov) [inline notificationdemo/source/c/CMakeLists.txt]: Done

File: datacollectiondemo/source/c/src/kaa_demo.c (hunk 7 above)
Comment (ohamov) [inline datacollectiondemo/source/c/src/kaa_demo.c]: Done

File: datacollectiondemo/source/c/CMakeLists.txt (hunk 6 above)
Comment (ohamov) [inline datacollectiondemo/source/c/CMakeLists.txt]: Agree with you. Also I think it is useful to keep build of sdk and sample app in separate directories.

**Comment by ohamov** [comment]
@rasendubi take a look please

**Comment by rasendubi** [comment]
LGTM

**Comment by forGGe** [comment]
+

**Pull request merged**